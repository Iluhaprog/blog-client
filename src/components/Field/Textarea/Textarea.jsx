import React from 'react';
import PropTypes from 'prop-types';
import {FieldWrapper} from '../Styled/FieldWrapper';

const TextareaField = ({
  field,
  form,
  meta,
  label,
  ...props
}) => {
  return (
    <FieldWrapper>
      <label>{label}</label>
      <textarea {...field} {...props}>{field.value}</textarea>
    </FieldWrapper>
  );
};

TextareaField.propTypes = {
  field: PropTypes.object,
  form: PropTypes.object,
  meta: PropTypes.object,
  label: PropTypes.string,
};

export {TextareaField};
