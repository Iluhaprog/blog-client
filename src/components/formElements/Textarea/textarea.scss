@import '../../../vars.scss';

.textarea {
    min-height: 130px;
    width: 100%;
    font-family: $main-font;
    font-size: 16px;
    box-shadow: $shadow;
    border-radius: 5px;
    margin-bottom: 20px;
    border: $border-2;
    padding: 15px;
    resize: none;
    outline: none;
    transition: .1s ease-in;

    &::placeholder {
        color: $text-color-4;
    }

    &:focus {
        border: $border-1;
    }
}