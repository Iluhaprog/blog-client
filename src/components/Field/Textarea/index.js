import {TextareaField} from './Textarea';
export {TextareaField};
