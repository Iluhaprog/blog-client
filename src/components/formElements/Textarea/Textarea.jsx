import React from 'react';
import { Field } from "redux-form";
import './textarea.scss';

export default props => (
    <>
        <Field
            className='textarea'
            name={props.name} 
            placeholder={props.placeholder || ''}
            component='textarea'
        />
    </>
);