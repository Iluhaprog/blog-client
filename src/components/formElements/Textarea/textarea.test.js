import React from 'react';
import { render } from '@testing-library/react';
import { reduxForm } from "redux-form";
import { Provider } from 'react-redux';
import configureMockStore from 'redux-mock-store';

import Textarea from './Textarea';

const mockStore = configureMockStore();


const decor = el => {
    const form = () => (<form>{el}</form>);
    return reduxForm({
        form: 'test',
    })(form);
}


test('Test view of textaria', () => {
    const store = mockStore({});
    const Ta = decor(
        <Textarea name='test_name' />
    );
    const { asFragment } = render(<Provider store={store}><Ta /></Provider>);
    expect(asFragment()).toMatchSnapshot();
});