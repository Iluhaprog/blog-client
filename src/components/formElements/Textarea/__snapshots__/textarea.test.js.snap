// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test view of textaria 1`] = `
<DocumentFragment>
  <form>
    <textarea
      class="textarea"
      name="test_name"
      placeholder="test"
    />
  </form>
</DocumentFragment>
`;
